src/search/ExactSearchQuery.jl

```diff
@@ -0,0 +1,200 @@
+"""
+Query type for exact sequence search.
+"""
+struct ExactSearchQuery{F<:Function,S<:BioSequence}
```

Comment:
Would be nice with a simple docstring for this type, e.g. how to construct it, maybe even doctests.